### ISSUE TITLE
Next major version

### ISSUE THREAD

**Issue by xyc** [storybookjs/react-inspector#50]
With the recent update of React: https://reactjs.org/blog/2018/03/29/react-v-16-3.html and updates from Chrome DevTools, it's time to update react-inspector to use the new context api and update it to match Chrome's changes.

Related issues:
- #47
- https://developers.google.com/web/updates/2017/05/devtools-release-notes#object-previews

`^3.0.0` will support only react>=16.3.0
`^2.0.0` supports older version of React.

**Comment by Hypnosphi** [comment]
You can keep supporting earlier versions using [`create-react-context`](https://github.com/jamiebuilds/create-react-context)

**Comment by xyc** [comment]
Yeah that's an option. But, `create-react-context` uses `componentWillReceiveProps` under the hood

**Comment by Hypnosphi** [comment]
I think that's OK, the component using this method will only be rendered in <=16.2

**Comment by xyc** [comment]
Make sense, I'll try it if it's stable enough for production

**Comment by Sidnioulz** [comment]
Hi folks,

Closing this issue as it's very old.
